File: cpp/gms/gms-filters-provider/src/descriptions/LinearIIRFilterDescription.cpp
#include "LinearIIRFilterDescription.hh"

LinearIIRFilterDescription::LinearIIRFilterDescription(IIRFilterParameters parameters,
                                                       bool causal,
                                                       std::string comments,
                                                       double highFrequency,
                                                       double lowFrequency,
                                                       int passBandType,
                                                       int filterDesignModel,
                                                       int order,
                                                       bool zeroPhase) : parameters(parameters),
                                                                        BaseLinearFilterDescription(causal,
                                                                                                    comments,
                                                                                                    highFrequency,
                                                                                                    lowFrequency,
                                                                                                    passBandType,
                                                                                                    filterDesignModel,
                                                                                                    order,
                                                                                                    zeroPhase){};

File: typescript/user-interface/packages/ui-wasm/src/ts/gms-filters-provider/gms-filters/types/linear-iir-filter-description.ts
import type { WasmFilterComputationType } from './filter-computation-type';
import type { WasmIIRFilterParameters } from './iir-filter-parameters';
import type { WasmLinearFilterDescription } from './linear-filter-description';

export interface WasmLinearIIRFilterDescriptionModule {
  new (
    parameters: WasmIIRFilterParameters,
    causal: boolean,
    comments: string,
    highFrequency: number,
    lowFrequency: number,
    passBandType: number,
    filterDesignModel: number,
    order: number,
    zeroPhase: boolean
  ): WasmLinearIIRFilterDescription;
}

export interface WasmLinearIIRFilterDescription extends WasmLinearFilterDescription {
  $$: {
    ptr: number;
  };
  delete: () => void;

  causal: boolean;
  comments: string;
  filterType: WasmFilterComputationType.IIR;
  parameters: WasmIIRFilterParameters;
}
